### DISCUSSION TITLE
Faciliter le reporting des travaux réalisés sur GitHub

### DISCUSSION THREAD

**Discussion by guillett** [betagouv/cap#4] [Besoin]
Lorsqu'il faut produire des livrables ou des services faits, il est possible d'utiliser la notion de _milestone_ sur GitHub pour regrouper les PR et les associer à un bon de commande précis. Le lien vers le _milestone_ pourra ensuite être indiqué dans le livrable pour expliciter ce qui a été fait.

Je me demande si de l'outillage pour être mis en place pour généraliser cette pratique.

**Comment by revolunet** [comment]
est-ce que ca aurait un rapport avec les "releases" GitHub ? 

je livre la 2.3.45, qu'est-ce qu'il y a dedans ? ex : https://github.com/betagouv/mon-entreprise/releases/tag/v1.28.0

**Reply by guillett** [reply]
Ah oui c'est vrai, on pourrait faire comme ça.

**Comment by guillett** [comment]
Je pensais à une version qui passe à l'échelle de ce qu'on a mis en place sur Aides Jeunes.

https://github.com/betagouv/aides-jeunes/milestone/18?closed=1
Pour un BC on a les tickets réalisés.

Aujourd'hui uniquement dans GH on a une GH action qui fait l'association automatiquement mais sur un seul dépôt.
https://github.com/betagouv/aides-jeunes/blob/main/.github/workflows/associate-actual-milestone.yml

Ça pourrait être cool de faire ça en multi dépôts et multi organisations.
Dans notre cas, on contribue à environ 5 dépôts régulièrement (et jusqu'à 9).
La liste des PR (mergées ou fermées) me parait être au bon niveau de reporting des BC.

**Reply by revolunet** [reply]
J'ai pas compris pourquoi faire une milestone dédiée ? la liste des PR suffit pas ?

**Reply by guillett** [reply]
Si mais une liste de PR c'est hyper compliquée à consolider sans milestone.

**Reply by guillett** [reply]
255 PR pour le milestone que j'ai partagé.

**Comment by revolunet** [comment]
J'ai publié ce script qui permet de dumper l'activité GitHub: https://github.com/revolunet/github-recap
